tomlparams/tests/test_tomlparams.py

```diff
@@ -653,6 +656,62 @@ def create_params():
 
         os.environ.pop('TOMLPARAMSCHECKING', None)
 
+    def test_concatenate_keys(self):
+        d = {'a': {'b': 1, 'c': 2}, 'd': 3}
+        expected = (('a.b', 1), ('a.c', 2), ('d', 3))
+        self.assertEqual(tuple(concatenate_keys(d)), expected)
+
+    def test_read_defaults_as_directory(self):
+        stddir = os.path.join(XDIR, 'tomlparams')
+        userdir = os.path.join(XDIR, 'usertomlparams')
+        defaults_as_dir = os.path.join(stddir, 'defaults_as_dir')
+        defaults_as_file = os.path.join(stddir, 'defaults_as_file')
+        params_default_as_dir = TOMLParams(
+            defaults_as_dir,
+            standard_params_dir=stddir,
+            user_params_dir=userdir,
+            verbose=False,
+        )
+        params_default_as_file = TOMLParams(
+            defaults_as_file,
+            standard_params_dir=stddir,
+            user_params_dir=userdir,
+            verbose=False,
+        )
+        self.assertEqual(params_default_as_dir, params_default_as_file)
+
+    @parameterized.expand(['human', 'animals', 'fungi'])
+    def test_content_defaults_as_directory(self, default_primary_key: str):
+        # primary keys of default as file
+        # primary_keys = ['human', 'animals', 'fungi']
+        stddir = os.path.join(XDIR, 'tomlparams')
+        userdir = os.path.join(XDIR, 'usertomlparams')
+        defaults_as_dir = os.path.join(stddir, 'defaults_as_dir')
+        defaults_as_file = os.path.join(stddir, 'defaults_as_file')
+        params_default_as_dir = TOMLParams(
+            defaults_as_dir,
+            standard_params_dir=stddir,
+            user_params_dir=userdir,
+            verbose=False,
+        )
+        params_default_as_file = TOMLParams(
+            defaults_as_file,
+            standard_params_dir=stddir,
+            user_params_dir=userdir,
+            verbose=False,
+        )
+        d_as_dir = tuple(
+            concatenate_keys(
+                params_default_as_dir[default_primary_key].get_params()
+            )
+        )
+        d_as_file = tuple(
+            concatenate_keys(
+                params_default_as_file[default_primary_key].get_params()
+            )
+        )
+        self.assertEqual(d_as_dir, d_as_file)
```

Comment:
Will order of keys matter here too in this comparison? If so, would we want the test to fail if key order differed? If not, might be worth comparing sets after `concatenate_keys()` here?